src/complex/Pipeline/AbstractPipelineNode.hpp

```diff
@@ -122,6 +125,16 @@ class COMPLEX_EXPORT AbstractPipelineNode
    */
   virtual bool preflight(DataStructure& data, const std::atomic_bool& shouldCancel) = 0;
 
+  /**
+   * @brief Attempts to preflight the node using the provided DataStructure.
+   * Returns true if preflighting succeeded. Otherwise, this returns false.
+   * @param data
+   * @param RenamedPaths& renamedPaths = {} Collection of renamed output paths.
+   * Only used for PipelineFilters.
+   * @return bool
+   */
+  virtual bool preflight(DataStructure& data, RenamedPaths& renamedPaths, const std::atomic_bool& shouldCancel) = 0;
```

Comment:
If `renamedPaths` is an output, should it be part of the return type?

Reply:
The output is the success state. The renamedPaths is a parameter that is shared through the course of preflighting a Pipeline.